frontend/src/app/modules/main/user-profile/user-profile.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnInit } from '@angular/core';
+import { Song } from 'src/app/models/song/song';
+import { SongsService } from 'src/app/services/songs/songs.service';
+
+@Component({
+  selector: 'app-user-profile',
+  templateUrl: './user-profile.component.html',
+  styleUrls: ['./user-profile.component.sass']
+})
+export class UserProfileComponent implements OnInit {
+  public isProfileMenuShown: boolean = false;
+
+  private readonly _scrollingSize: number = 270;
+  public topArtists = new Array<object>();
```

Comment:
Instead of  new Array<object>(); should be something like Artist[]. Use interface instead of an object.